src/libraries/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Hosting/CompositionServices.cs

```diff
@@ -307,10 +307,10 @@ internal static void TryExportMetadataForMember(this MemberInfo member, out IDic
 
                             if (reservedMetadataNames.Contains(pi.Name, StringComparers.MetadataKeyNames))
                             {
-                                throw ExceptionBuilder.CreateDiscoveryException(SR.Discovery_ReservedMetadataNameUsed, member.GetDisplayName(), provider.Name);
```

Comment:
This is the case where `provider == null` seems this exception handling copied from other case but forgotten to change variable names accordingly, please review @safern @stephentoub